tesseract/tesseract_planning/tesseract_motion_planners/test/ompl_planner_tests.cpp

```diff
@@ -304,6 +303,91 @@ TEST(OMPLMultiPlanner, OMPLMultiPlannerUnit)  // NOLINT
   ompl_config->end_waypoint = std::make_shared<tesseract_motion_planners::JointWaypoint>(ewp, kin->getJointNames());
 }
 
+#ifndef OMPL_LESS_1_4_0
+
+class GlassUprightConstraint : public ompl::base::Constraint
+{
+public:
+  GlassUprightConstraint(const Eigen::Vector3d& normal, tesseract_kinematics::ForwardKinematics::Ptr fwd_kin)
+    : ompl::base::Constraint(fwd_kin->numJoints(), 1), normal_(normal), fwd_kin_(std::move(fwd_kin))
+  {
+  }
+
+  ~GlassUprightConstraint() override = default;
+
+  void function(const Eigen::Ref<const Eigen::VectorXd>& x, Eigen::Ref<Eigen::VectorXd> out) const override
+  {
+    Eigen::Isometry3d pose;
+    fwd_kin_->calcFwdKin(pose, x);
+
+    Eigen::Vector3d z_axis = pose.matrix().col(2).template head<3>().normalized();
+
+    out[0] = z_axis.dot(normal_);
```

Comment:
I think your function here should actually set `out[0] = 1.0 - z_axis.dot(normal_)` such when the vectors are aligned (dot product = 1.0), your return value is zero

Reply:
Fixed.